### ISSUE TITLE
Bluetooth file transfer

### ISSUE THREAD

**Issue by lukew3** [localsend/localsend#144]
### Is your feature request related to a problem? Please describe.

When I’m on my universities network, I cannot transfer files between my devices. I’m not completely sure why this is, but I’d guess that there is some filter in place to prevent devices from being visible to each other. Also, if you have devices that are not on the same network or you are both offline, transfer is not possible.

### Describe the solution you'd like

I’d like a solution where devices can interact with each other over Bluetooth. It would be nice if local share could open itself up to bluetooth connections and listen for other localShare instances. However, if this is not possible, maybe you could list devices connected with Bluetooth and have an option to send data to devices if they have LocalSend installed.

### Describe alternatives you've considered

Bluetooth file transfer is already possible natively on android devices, but does not exist from apple to android. I’m not sure about between other systems.

**Comment by electr1fy0** [comment]
Pretty sure macOS and Android can transfer files using Bluetooth already. _(Source: I just did it.)_
No idea about iOS.
It would be cool to share text though.

**Comment by GameOverFlowChart** [comment]
Sending files over Bluetooth is something all computers with Bluetooth (smartphones included) can already do, this would only complicate the ui of the app and bring no new functionality.

**Comment by electr1fy0** [comment]
The UI is something that already needs quite a bit of improvement and adding more options will only bloat it further, I agree.

**Comment by lukew3** [comment]
> Sending files over Bluetooth is something all computers with Bluetooth (smartphones included) can already do, this would only complicate the ui of the app and bring no new functionality.

Transferring files via Bluetooth the normal way requires you connect each device to another with Bluetooth through the system settings and then going to the file in the files app, select share to  Bluetooth, select the device, and then share it. I think that localsend could condense this into one step that is performed automatically by localsend, without adding much more to the UI. You could just have a small indicator on the device name to say whether it is a Bluetooth or WiFi connection.

**Comment by GameOverFlowChart** [comment]
> > Sending files over Bluetooth is something all computers with Bluetooth (smartphones included) can already do, this would only complicate the ui of the app and bring no new functionality.
> 
> Transferring files via Bluetooth the normal way requires you connect each device to another with Bluetooth through the system settings and then going to the file in the files app, select share to Bluetooth, select the device, and then share it. I think that localsend could condense this into one step that is performed automatically by localsend, without adding much more to the UI. You could just have a small indicator on the device name to say whether it is a Bluetooth or WiFi connection.
> 
> If the goal of this project is to provide a replacement to airdrop, it should work similarly to airdrop, which sends data over Bluetooth.

I'm pretty sure that I have send files by enabling Bluetooth on both devices, then selecting a file -> share menue -> Bluetooth -> selecting the device other device -> accepting the send request on the other device. So no connecting the devices over system settings isn't required according to my memory and if it would be a required step than I don't think that it would be technically possible for an app to skip that step (without root at least). But the Bluetooth standard evolved a lot over the years so there could be something that I'm missing.

**Comment by lukew3** [comment]
> I'm pretty sure that I have send files by enabling Bluetooth on both devices, then selecting a file -> share menue -> Bluetooth -> selecting the device other device -> accepting the send request on the other device. So no connecting the devices over system settings isn't required according to my memory and if it would be a required step than I don't think that it would be technically possible for an app to skip that step (without root at least). But the Bluetooth standard evolved a lot over the years so there could be something that I'm missing.

Sorry, you're right, you don't have to connect via system settings. It seems that the [flutter_blue](https://github.com/pauldemarco/flutter_blue) library should be able to handle everything needed to discover clients, connect, and send files without root.

**Comment by Bronts** [comment]
In my company, desktop office computers are only connected to the corporate Intranet, and wireless cards are not allowed, but separate laptops are allowed to access the Internet to look up some documents. Sometimes it is necessary to transfer some documents under a few hundred KB between the laptop and the desktop computer, and the use of the USB disk is more troublesome. So using Bluetooth should be more convenient. In addition, Windows built-in Bluetooth transfer can be used for file transfer, but the operation is cumbersome. It is hoped that the LocalSend team will fully consider such use cases.

**Comment by p0358** [comment]
Yeah it's hardly an AirDrop alternative like description claims, when AirDrop doesn't need connecting to a common Wi-Fi or anything like that... You just select a nearby device, accept on the other side and off it goes...

> Sending files over Bluetooth is something all computers with Bluetooth (smartphones included) can already do, this would only complicate the ui of the app and bring no new functionality.

Haha. Not on iOS.

**Comment by phith0n** [comment]
Looking forward to the bluetooth supportion because my company block the internal network sharing for security purpose.

And most of time I'm sending and receiving text rather than files, so it's troublesome to use bluetooth that provided by system.

**Comment by rpc31** [comment]
Same! If that feature's up, that'd be really great!

**Comment by MMachado05** [comment]
I'm glad there are other people voting for this feature! I hope this gets a good look-at by someone with the know-how to make a PR with this implemented.

**Comment by GameOverFlowChart** [comment]
> Haha. Not on iOS.

Sounds like there is a need for a separate iOS app that adds this basic functionality to iPhones which even ancient Nokia and Sony Ericsson phones have. But I don't see a reason why that functionality should be stacked on top of Localsend if most devices already have that build in. I'm blindly assuming that you are right and iPhones are really missing this ancient feature, but that makes me doubt even more why anyone would buy such devices.

**Comment by p0358** [comment]
Trust me, I was quite a bit surprised at first too... Same reason why iMessage was never brought to Android or RCS to iOS. They do this to keep their monopoly in the USA, where it's some kind of social stigma to not own an iPhone apparently in some circles due to these.

Also there used to be a tweak for old jailbroken iOS to add this feature (OBEX BT file sharing), but then that's kinda offtopic here (it was called AirBlue Sharing)

**Comment by MMachado05** [comment]
I randomly stumbled upon something called [Berty](https://berty.tech/) which is a no-internet instant messenger that uses Bluetooth Low Energy to support its messaging. Maybe this could be looked at for how Bluetooth file transfer in Localsend might be accomplished?
